syz-verifier/main.go

```diff
@@ -267,6 +276,70 @@ func (srv *RPCServer) NextExchange(a *rpctype.NextExchangeArgs, r *rpctype.NextE
 	return nil
 }
 
+// newResult is called when a Runner sends a new Result. It returns true if all
+// Results from the corresponding programs have been received and they can be
+// sent for verification. Otherwise, it returns false.
+func (srv *RPCServer) newResult(res *verf.Result, idx int) bool {
+	for _, prog := range srv.progs {
+		if prog.idx == idx {
+			prog.res = append(prog.res, res)
+			delete(prog.left, res.Pool)
+			return len(prog.left) == 0
+		}
+	}
+	return false
+}
+
+// processResults will send a set of complete results for verification and, in
+// case differences are found, it will store a result report highlighting those
+// in th workdir/results directory. If writing the results fails, it returns an
+// error.
+func (vrf *Verifier) processResults(res []*verf.Result, prog *prog.Prog) {
+	rr := verf.Verify(res, prog)
+	if rr == nil {
+		return
+	}
+	data := fmt.Sprintf("Errno mismatches found for program:\n%s\n", rr.Prog)
+	data += "CALL REPORTS FOUND BELOW\n"
+
+	for _, cr := range rr.Reports {
+		data += fmt.Sprintf("Report for call: %s", cr.Call)
+
+		if cr.Mismatch {
+			data += " - MISMATCH FOUND"
+		}
+		data += "\n"
+		for pool, errno := range cr.Errnos {
+			data += fmt.Sprintf("Pool: %d, Errno: %d\n", pool, errno)
+		}
+		data += "\n"
+	}
+
+	oldest := 0
+	var oldestTime time.Time
+	for i := 0; i < maxResultReports; i++ {
+		info, err := os.Stat(filepath.Join(vrf.resultsdir, fmt.Sprintf("result-%d", i)))
+		if err != nil {
+			// There are only i-1 report files so the i-th one
+			// can be created.
+			oldest = i
+			break
+		}
+
+		// Otherwise, search for the oldest report file to
+		// overwrite as newer result reports are more useful.
+		if oldestTime.IsZero() || info.ModTime().Before(oldestTime) {
+			oldest = i
+			oldestTime = info.ModTime()
+		}
+	}
+
+	err := osutil.WriteFile(filepath.Join(vrf.resultsdir, fmt.Sprintf("result-%d", oldest)), []byte(data))
+	if err != nil {
+		log.Printf("failed to write result-%d file, err %v", oldest, err)
+	}
+}
```

Comment:
It may be useful to print something to console as well so that user knows that there is a new file to look at.